cs3api4lab/api/cs3_ocm_share_api.py

```diff
@@ -147,21 +144,28 @@ def update_received(self, ocm_share_id, field, value):
             self._handle_error(response, "Error updating OCM received share:")
         return response.opaque
 
-    def list(self, share_id=None):
-        if share_id is None:
-            return self._list_ocm_shares()
-        else:
-            return self._get_ocm_share(share_id)
+    def list(self, optional_filter={}):
```

Comment:
> I guess this way is to retrieve just a specific resource instead of a list with all of them?

Yes, this is exactly what it does. I have noticed that in a lot of places in our plugin we are not using filters implemented by Reva API. Most of the time we are getting a whole shared file list (even when there is a need only for information on one file) for a user and then filtering them by a path, that design is heavy on load and is not "bulletproof". By using filters we can cut down the load and receive from Reva only shares that we need, especially that shared files are identified by resource id (storage_id, opaque_id). This also resolves some problems like how to determine share and OCM, previous implementation was bugged.